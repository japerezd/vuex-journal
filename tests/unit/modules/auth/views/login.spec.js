import { shallowMount } from '@vue/test-utils';
import Login from '@/modules/auth/views/Login.vue';
import createVuexStore from '../../../mock-data/mock-store';
import Swal from 'sweetalert2';

jest.mock('sweetalert2', () => ({
  fire: jest.fn(),
  showLoading: jest.fn(),
  close: jest.fn(),
}));

const mockRouter = {
  push: jest.fn(),
};

jest.mock('vue-router', () => ({
  useRouter: () => mockRouter,
}));

describe('Pruebas en el login component', () => {
  const store = createVuexStore({
    status: 'not-authenticated',
    user: null,
    idToken: null,
    refreshToken: null,
  });

  store.dispatch = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
  });

  test('debe hacer match con snapshot', () => {
    const wrapper = shallowMount(Login, {
      global: {
        plugins: [store],
        stubs: ['router-link'],
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('credenciales incorrectas, disparar error del SWAL', async () => {
    store.dispatch.mockReturnValueOnce({
      ok: false,
      message: 'Error en credenciales',
    });

    const wrapper = shallowMount(Login, {
      global: {
        plugins: [store],
        stubs: ['router-link'],
      },
    });

    await wrapper.find('form').trigger('submit');
    expect(store.dispatch).toHaveBeenCalledWith('auth/signInUser', {
      email: '',
      password: '',
    });

    expect(Swal.fire).toHaveBeenCalledWith(
      'Error',
      'Error en credenciales',
      'error'
    );
  });

  test('debe de redirigir a la ruta no-entry', async () => {
    store.dispatch.mockReturnValueOnce({
      ok: true,
    });

    const wrapper = shallowMount(Login, {
      global: {
        plugins: [store],
        stubs: ['router-link'],
      },
    });

    const [txtEmail, txtPassword] = wrapper.findAll('input');
    await txtEmail.setValue('beto@gmail.com');
    await txtPassword.setValue('123456');

    await wrapper.find('form').trigger('submit');
    expect(store.dispatch).toHaveBeenCalledWith('auth/signInUser', {
      email: 'beto@gmail.com',
      password: '123456',
    });

    expect(mockRouter.push).toHaveBeenCalledWith({ name: 'no-entry' })
  });
});
