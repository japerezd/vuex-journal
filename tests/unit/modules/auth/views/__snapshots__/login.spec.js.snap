// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el login component debe hacer match con snapshot 1`] = `
<span class="login100-form-title p-b-41"> Ingresar </span>
<form class="login100-form validate-form p-b-33 p-t-5">
  <div class="wrap-input100 validate-input" data-validate="Enter email"><input class="input100" type="email" placeholder="Correo" required=""><span class="focus-input100" data-placeholder=""></span></div>
  <div class="wrap-input100 validate-input" data-validate="Enter password"><input class="input100" type="password" placeholder="Contraseña" required=""><span class="focus-input100" data-placeholder=""></span></div>
  <div class="container-login100-form-btn m-t-32"><button type="submit" class="login100-form-btn"> Login </button></div>
  <div class="container-login100-form-btn m-t-32">
    <router-link-stub to="[object Object]"></router-link-stub>
  </div>
</form>
`;
